plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.plugin.catalog;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.security.PrivilegedExceptionAction;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.sql.catalog.CatalogService;
+import org.opensearch.sql.opensearch.security.SecurityAccess;
+import org.opensearch.sql.opensearch.setting.OpenSearchSettings;
+import org.opensearch.sql.storage.StorageEngine;
+
+/**
+ * This class manages catalogs and responsible for creating connectors to these catalogs.
+ */
+public class CatalogServiceImpl implements CatalogService {
+
+
+  private final Set<String> allowedConnectors = new HashSet<>() {{
+      add("prometheus");
```

Comment:
any architecture diagram / docs for catalog? looks like it's assumed `opensearch` if there are no connectors, but only `prometheus` is the allowed connector. if we want to make opensearch default then `source=table` and `source=opensearch.table` should both work?

Reply:
Good Question. Will prepare docs in the last PR.
This is my mental model.
Users will only use catalogs which are explicitly mentioned in the catalog.json configuration file. 

For commands without using any catalog mentioned in the configuration, I am going to treat it as an OpenSearch Index and fire query on the index.

For example:
with catalog configuration containing a catalog with name `prometheus1`


1)  `source=table` I am going to fire query on index `table` in the OS instance.

2)  `source=opensearch.table `I am going to fire query on index `opensearch.table` in the OS instance.
3)  `source=prometheus.http_requests_table`.I am going to fire query on index `prometheus.http_requests_table` in the OS instance.

4)  `source=prometheus1.http_requests_table`
 I am going to fire query on metric `http_requests_table `in the prometheus1 instance.

Reply:
do we plan add opensearch connector later?

Reply:
Yes need to create an issue for this. Then we could connect to multiple instances